src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -185,6 +185,10 @@ protected function validateType(&$value, $type, $coerce = false)
         }
 
         if ('array' === $type) {
+            if ($coerce) {
+                $value = $this->toArray($value);
```

Comment:
Well, I think this is the issue that spawned the feature: https://github.com/epoberezkin/ajv/issues/158

They discuss it a bit, then come up with the `coerceTypes:'array'` idea